Backport of patch from upstream tree, to build against calamares 3.2.43:
https://github.com/calamares/calamares-extensions/commit/718f73c9c862995163ae39c79f1725f3357fcf91

From 2ac6b56c58e1ca3376746a15ac4e610c20082934 Mon Sep 17 00:00:00 2001
From: Adriaan de Groot <groot@kde.org>
Date: Thu, 9 Sep 2021 10:57:21 +0200
Subject: [PATCH] CMake: adjust extensions to changes in Calamares core

(cherry picked from commit 718f73c9c862995163ae39c79f1725f3357fcf91)
---
 CMakeLists.txt                       | 7 +++----
 modules/mobile/MobileQmlViewStep.cpp | 8 +++-----
 2 files changed, 6 insertions(+), 9 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 795c7cb..4166fb9 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -46,13 +46,12 @@ project(calamares-extensions
 # consumers by loading the developer's config from a build
 # directory (which doesn't have the rest of the config
 # installed inside it).
-find_package(Calamares 3.2.26 REQUIRED NO_CMAKE_PACKAGE_REGISTRY)
+set( CALAMARES_VERSION_REQUIRED 3.2.43 )
+find_package(Calamares ${CALAMARES_VERSION_REQUIRED} NO_CMAKE_PACKAGE_REGISTRY)
 if (NOT TARGET Calamares::calamares OR NOT TARGET Calamares::calamaresui)
-    message(FATAL_ERROR "Calamares found with missing CMake targets")
+    find_package(Calamares ${CALAMARES_VERSION_REQUIRED} REQUIRED)
 endif()
 
-find_package(YAMLCPP REQUIRED)  # Needed to untangle some dependencies before Calamares 3.2.36
-
 ### CMAKE SETUP
 #
 # Enable IN_LIST
diff --git a/modules/mobile/MobileQmlViewStep.cpp b/modules/mobile/MobileQmlViewStep.cpp
index 8b42f39..a4ac96f 100644
--- a/modules/mobile/MobileQmlViewStep.cpp
+++ b/modules/mobile/MobileQmlViewStep.cpp
@@ -2,16 +2,14 @@
  * SPDX-License-Identifier: GPL-3.0-or-later */
 #include "MobileQmlViewStep.h"
 
+#include "Branding.h"
 #include "GlobalStorage.h"
-
-#include "locale/LabelModel.h"
+#include "locale/TranslationsModel.h"
+#include "modulesystem/ModuleManager.h"
 #include "utils/Dirs.h"
 #include "utils/Logger.h"
 #include "utils/Variant.h"
 
-#include "Branding.h"
-#include "modulesystem/ModuleManager.h"
-
 #include <QProcess>
 
 CALAMARES_PLUGIN_FACTORY_DEFINITION( MobileQmlViewStepFactory, registerPlugin< MobileQmlViewStep >(); )
-- 
2.30.2

